test/test_pipeline_schedule.py

```diff
@@ -320,38 +317,33 @@ def test_interleaved_1f1b(self):
         microbatches = [
             (torch.randn_like(microbatch),) for _ in range(num_microbatches)
         ]
+        target_mbs = [
+            torch.randn_like(microbatch) for _ in range(num_microbatches)
+        ]
 
+        loss_fn = torch.nn.MSELoss()
         schedule = ScheduleInterleaved1F1B(
             stages,
             num_microbatches,
+            loss_fn=loss_fn,
         )
-        schedule.step_microbatches(microbatches)
+        schedule.step_microbatches(microbatches, target_mbs=target_mbs)
```

Comment:
Should we feed target to every rank?

Reply:
we don't have to, but if we do that is also allright too since if it is not the last stage then it wont calculate the loss using the target_mbs.

Reply:
I'd rather not rely on assumptions :) 
We should test the real case (strictly speaking even `microbatches` should not be fed to every rank).